src/dbnode/persist/fs/retriever.go

```diff
@@ -434,18 +435,17 @@ func (r *blockRetriever) fetchBatch(
 	reqs := r.filterAndCompleteWideReqs(allReqs, seeker, seekerResources,
 		retrieverResources)
 
-	var limitErr error
-	if err := r.queryLimits.AnyExceeded(); err != nil {
-		for _, req := range reqs {
+	for _, req := range reqs {
+		if err := r.queryLimits.AnyExceeded(); err != nil {
 			req.err = err
+			continue
```

Comment:
Hm instead of continuing here, can we just set req.err on all of them in a for loop and be done? Worried about it taking longer to abort of all them if global limits are tripped.

Reply:
ahhh that's what that was doing. whoops sorry for changing that.